### ISSUE TITLE
[Android] Start foreground service on boot?

### ISSUE THREAD

**Issue by ryanbrink** [notifee/react-native-notifee#360]
I'm using the foreground service functionality, but I'd like it to start on boot and not require the user to open the App to launch it. I don't see anything in the docs, and my attempts to create something that receives the `BOOT_COMPLETED` action to start the Notifee service don't seem to work.

**Comment by helenaford** [comment]
Oh, interesting, do you have a use case to help understand why this is needed?

My initial thought is that you're displaying the notification when the app is in a headless state (quit) and need the notification to be foreground service 🤔

**Comment by ryanbrink** [comment]
Good point: in my case it's because of location tracking, which isn't directly related to Notifee. However, as an App can only have a single foreground service at a time I'd like to have it launched on boot, even though there may be nothing notification-related to handle. 

Background notifications do work as you mentioned.

**Comment by mikehardy** [comment]
Interesting inter-operation case. I also use react-native-background-geolocation-android in one my projects, and one thing I notice there is that they have a static id for their android notification. The implication is that you may then post to that notification if need be, for inter-operation cases like this.

I don't believe that is a formal feature of the library there but if my understanding is correct it does appear to be quietly supported.

Is that the library you use? That's mostly curiosity.

I think on our end we could perhaps support a feature that allowed you to specify the ID of the foreground service notification so that the "only single foreground service notification at a time" constraint is met?

To be clear this is a pretty vague sketch - I haven't looked into this restriction or thought through this use case much - it's based on vague recollections - but if I'm on target perhaps that would work.

If I'm totally off-target (just say so!) then more research is warranted
